app/controllers/houseConsignment/index/items/document/TypeController.scala

```diff
@@ -118,5 +118,13 @@ class TypeController @Inject() (
     for {
       updatedAnswers <- Future.fromTry(request.userAnswers.set(TypePage(houseConsignmentIndex, itemIndex, documentIndex), value))
       _              <- sessionRepository.set(updatedAnswers)
-    } yield Redirect(navigator.nextPage(TypePage(houseConsignmentIndex, itemIndex, documentIndex), mode, request.userAnswers))
+      redirection = value.`type` match {
+        case DocType.Support =>
+          controllers.houseConsignment.index.items.document.routes.AddAdditionalInformationYesNoController
+            .onPageLoad(request.userAnswers.id, mode, houseConsignmentIndex, itemIndex, documentIndex)
+        case _ =>
+          controllers.houseConsignment.index.items.document.routes.AddAnotherDocumentController
+            .onPageLoad(request.userAnswers.id, houseConsignmentIndex, itemIndex, mode)
+      }
+    } yield Redirect(redirection)
```

Comment:
Can this logic be moved to the DocumentNavigator?

Reply:
Yeah maybe, I kept it here because I already had the DocType (`value.`type``), without having to do another get in the navigator

Reply:
Negligible performance hit to pick that value out of user answers. We already have user answers in memory so it's not as if we're having to re-fetch it from mongo. It's a standard HRMC pattern.